### DISCUSSION TITLE
Lesson 6: Error while connecting meta mask to localhost

### DISCUSSION THREAD

**Discussion by sahilbansal123-lang** [Cyfrin/foundry-full-course-cu#684] [General]
I'm also getting this error while trying to connect to ganache

Command 
```
forge create SimpleStorage --rpc-url HTTP://127.0.0.1:7545 --interactive

```
AND ERROR im getting on terminal is: 

```

error sending request for url (http://127.0.0.1:7545/): error trying to connect: tcp connect error: Connection refused (os error 111)

Context:
- Error #0: error trying to connect: tcp connect error: Connection refused (os error 111)
- Error #1: tcp connect error: Connection refused (os error 111)
- Error #2: Connection refused (os error 111)

```

This is my ganache account which shows port 7545 with id 5777 and when im trying to add this to metamask for local host connection it shows wrong id

![Screenshot 2023-08-22 000704](https://github.com/Cyfrin/foundry-full-course-f23/assets/72665907/79b8a76a-6948-4db3-b04b-d9d571e51a95)

**Comment by devmousa** [comment]
do you face the same problem when you run it with ```anvil``` ?
I don't use ganache (because it needs MacOS to run it), but I'm using ```anvil``` and it works very well with me

just a question, isn't there a button to turn on or turn off ganache ?
try to rerun the app and rerun the command
and try to write ```HTTP``` in lowercase, maybe it works (it should be the same, but try it 😅)

good luck and happy coding ;)

**Reply by sahilbansal123-lang** [reply]
No bro not with Anvil. only facing it while working with ganache

**Comment by Shoolapani** [comment]
Hey Sahil, I also faced the same issue if you using WSL, try to change the server & select wsl


![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/83538929/2df9180b-357a-4268-847b-dddcebe0081d)


```
forge create SimpleStorage --rpc-url http://172.27.96.1:7545 --interactive
```

use small http
